traits/observers/tests/test_named_trait_observer.py

```diff
@@ -0,0 +1,87 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+import unittest
+from unittest import mock
+
+from traits.observers._named_trait_observer import (
+    NamedTraitObserver,
+)
+from traits.observers._observer_path import ObserverPath
+
+
+class TestNamedTraitObserver(unittest.TestCase):
+
+    def test_not_equal_notify(self):
+        observer1 = NamedTraitObserver(name="foo", notify=True)
+        observer2 = NamedTraitObserver(name="foo", notify=False)
+        self.assertNotEqual(observer1, observer2)
+        self.assertNotEqual(hash(observer1), hash(observer2))
```

Comment:
This isn't a safe check: it's possible for the hashes to be equal even though the objects aren't, and even if they're not equal in this version of Python, they could become so if the internal hash algorithm changes in a future version.

Reply:
Yep, equal-returns-true means hashes-are-equal but the reverse isn't true 🤦